public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx

```diff
@@ -115,28 +101,24 @@ export class ConfigEditor extends PureComponent<Props, State> {
 
   logAnalyticsHasNecessaryCredentials = () => {
     if (
-      !this.state.config.secureJsonFields.logAnalyticsClientSecret &&
-      !this.state.config.secureJsonData.logAnalyticsClientSecret
+      !this.props.options.secureJsonFields.logAnalyticsClientSecret &&
+      !this.props.options.secureJsonData.logAnalyticsClientSecret
     ) {
       return false;
     }
 
-    if (!this.state.config.jsonData.logAnalyticsClientId || !this.state.config.jsonData.logAnalyticsTenantId) {
+    if (!this.props.options.jsonData.logAnalyticsClientId || !this.props.options.jsonData.logAnalyticsTenantId) {
       return false;
     }
 
     return true;
   };
 
-  onConfigUpdate = (config: any) => {
-    this.updateDatasource(config);
-  };
-
   onLoadSubscriptions = async (type?: string) => {
-    await this.backendSrv.put(`/api/datasources/${this.state.config.id}`, this.state.config).then(() => {
-      this.updateDatasource({
-        ...this.state.config,
-        version: this.state.config.version + 1,
+    await this.backendSrv.put(`/api/datasources/${this.props.options.id}`, this.props.options).then(() => {
+      this.onOptionsUpdate({
+        ...this.props.options,
+        version: this.props.options.version + 1,
```

Comment:
This looks super strange, why is the data source version updated on a load action here? The version is updated in the backend DB code and should not be incremented by the frontend